packages/pkh-webauthn/src/index.ts

```diff
@@ -0,0 +1,221 @@
+    const credential = await credentials.create(opts)
+    if (!credential) throw new Error('Empty Credential Response')
+    // @ts-ignore CredentialsContainer does contain response
+    const { response } = credential
+    const authenticatorData = getAuthenticatorData(response)
+    const { publicKey } = decodeAuthenticatorData(authenticatorData)
+    return encodeDIDFromPub(publicKey)
+
+  }
+
+  export async function getAuthMethod (didOpts: {
+    did?: string
+    dids?: Array<string>,
+    selectDID?: DIDSelector
+  }): Promise<AuthMethod> {
+    const { did, dids } = didOpts
+
+    let selectDID: DIDSelector
+
+    if (didOpts.selectDID) selectDID = didOpts.selectDID // Use callback
+    else if (did) selectDID = async () => did // Use known DID
+    else if (dids) selectDID = async (a, b) => dids.find(x => x == a || x == b) // Use probe result
+    else throw new Error('getAuthMethod({ did|dids|selectDID }) expects one resolution option')
+
+    return async (opts: AuthMethodOpts): Promise<Cacao> => {
+      return createCacao(opts, selectDID)
+    }
+  }
+
+  async function createCacao (opts: AuthMethodOpts, selectDID: DIDSelector): Promise<Cacao> {
+    const now = new Date()
+    // The public key is not known at pre-sign time.
+    // so we sign a "challenge"-block without Issuer attribute
+    const challenge: Cacao = {
+      h: {
+        t: 'caip122'
+      },
+      p: {
+        domain: globalThis.location.hostname,
+        aud: opts.uri || globalThis.location.toString(),
+        version: '1',
+        nonce: opts.nonce || u8a.toString(randomBytes(8), 'base64url'),
+        resources: opts.resources,
+        exp: opts.expirationTime || new Date(now.getTime() + 7 * 86400000).toISOString(), // 1 week
+        nbf: opts.notBefore || now.toISOString(),
+        iat: opts.issuedAt || now.toISOString()
+      } as Payload // squelch missing p.iss
+    }
+
+    const block = await blockFromCacao(challenge) // await CacaoBlock.fromCacao(challenge) when issue resolved
+    // perform sign
+    const { credential, recovered } = await authenticatorSign(block.cid.bytes)
+
+    // @ts-ignore - credential.response does exist.
+    const { response } = credential
+    const { clientDataJSON, signature } = response
+    const authData = getAuthenticatorData(response)
+    const aad = assertU8(encode({ authData, clientDataJSON }))
+
+    const recoveredDIDs = recovered.map(encodeDIDFromPub)
+
+    const iss = await selectDID(recoveredDIDs[0], recoveredDIDs[1])
+    if (!iss) throw new Error('PublicKeySelectionFailed')
+    // Assert that the resolved key belongs to the signature
+    if (!recoveredDIDs.includes(iss)) throw new Error('UnrelatedPublickey')
+
+    // Insert iss + aad after signature
+    return {
+      h: challenge.h,
+      p: {
+        ...challenge.p,
+        iss,
+      },
+      s: {
+        t: 'webauthn:p256',
+        s: u8a.toString(signature, 'base64url'),
+        aad
```

Comment:
I think that any additional data should be part of `m` (SignatureMeta struct).
```suggestion
        m: { aad }
```